src/sys/unix/selector/io_source/edge_triggered.rs

```diff
@@ -0,0 +1,104 @@
+use crate::sys::Selector;
+use crate::{Interest, Registry, Token};
+use std::io;
+use std::os::unix::io::RawFd;
+
+struct InternalState {
+    selector: Selector,
+    token: Token,
+    interests: Interest,
```

Comment:
I think this can be `Option<Interest>`, where `None` means not registered and `Some` means registered, allows us to drop the `is_registered` boolean.

Reply:
For `RawFd` we can use -1, with a note saying it can be invalid if `interests` is `None`.